client/src/utils/Store.js

```diff
@@ -0,0 +1,162 @@
+import axios from 'axios';
+import clonedeep from 'lodash.clonedeep';
+import baseurl from '@/env';
+import {
+  set,
+  keys,
+  del,
+  get,
+} from 'idb-keyval';
+
+axios.defaults.withCredentials = true;
+export default class API {
+  constructor() {
+    this.plot = '';
+    this.init();
+  }
+
+  /**
+   * initializes cache
+   */
+  init = async () => {
+    const k = await keys();
+    this.models = new Map();
+    k.forEach((key) => {
+      this.models.set(key, get(key));
+    });
+    await Promise.all(this.models);
+  }
+
+  /**
+   * Flattens model to make request params
+   * @param {Object} model model to flatten
+   * @returns {Object} flattened params
+   */
+  flatten = (model) => {
+    const params = {};
+    Object.values(model).forEach((value) => {
+      Object.assign(params, value);
+    });
+    return params;
+  }
+
+  /**
+   * Gets plot from server
+   * @param {String} fig_type, type of figure to be requested from server
+   * @return {String} image data base64 string, or null if request fails
+   */
+  createPlot = async (fig_type = 'dndm') => {
+    // may need to call createModel on all items before getting plot
+    try {
+      const { data } = await axios.post(`${baseurl}/plot`, {
+        fig_type,
+        img_type: 'png',
+      });
+      this.plot = `data:image/png;base64,${data.figure}`;
+      return this.plot;
+    } catch (error) {
+      console.error(error);
+      return null;
+    }
+  }
```

Comment:
Would be good if this was called any time state updates (create/update/clone/delete) instead of having to lazily request it. Plot choice isn't specific to each model so this could have a home, here, in state as a global context.

Reply:
createPlot is called in create, update, and clone. I dont know what you mean here?

Reply:
If so then we don't need this right?

` // should redirect to home. Image should be requested when at home component \
this.image = await this.$store.createPlot(this.meta.fig_type);`